fission-web-server/library/Fission/Web/Server/Models.hs

```diff
@@ -99,6 +99,17 @@ UserChallenge
 
   deriving Show Eq
 
+--------------------------------------------------------------------------------
+
+UserRecoveryChallenge
+  userId     UserId
+  hash       Challenge
+  insertedAt UTCTime
+
+  UniqueRecoveryChallenge hash
+
+  deriving Show Eq
```

Comment:
Is this different from a `UserChallenge`?

Reply:
It's structurally not much different, it's only structural difference is an additional `insertedAt` field for tracking when the challenge expires.

But it's generally meant to be used for another purpose, and I think should therefore be its own table: The recovery process should run independently of the email verification process for accounts.
The biggest semantic difference is that recovery challenges are deemed expired after one hour. This is not the case for `UserChallenge`s.

Reply:
Ah, expiry, good point 👍